compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/providers/impl/FirTypeResolverImpl.kt

```diff
@@ -92,14 +96,34 @@ class FirTypeResolverImpl(private val session: FirSession) : FirTypeResolver() {
         }
     }
 
+    private fun FirQualifierResolver.resolveEnumSymbol(
+        qualifier: List<FirQualifierPart>,
+        classId: ClassId
+    ): FirVariableSymbol<FirEnumEntry>? {
+        val enumClassSymbol = resolveSymbolWithPrefix(qualifier.dropLast(1), classId) ?: return null
```

Comment:
Assuming the current `qualifier` refers to enum entry, we want to look up enum class so `dropLast(1)`? If so, add a comment please. If not, please explain with a comment (:D)

Reply:
Yep that's exactly what this is doing. Let me add a comment